### ISSUE TITLE
Add ability to detect Gutenberg inside render_callback

### ISSUE THREAD

**Issue by paulschreiber** [WordPress/gutenberg#23810]
**Is your feature request related to a problem? Please describe.**
Currently, it is difficult to detect if you are inside the editor (vs rendering a page) inside `render_callback`. This has been the case since at least 2018 (#8846).

To get the property, you have to write code like so:
```php
$is_backend = defined('REST_REQUEST') && true === REST_REQUEST && 'edit' === filter_input( INPUT_GET, 'context', FILTER_SANITIZE_STRING );
```

This is long, clunky and fragile. Individual and plugin implementors should not have to copy and paste this code.

**Example use cases**
* Show placeholder text in the backend, even when a value is missing.
* Disable form fields in the backend, preventing accidental submission.

**Describe the solution you'd like**
Create a helper function like so:
```php
	/**
	 * Whether or not we are rendering a block on the backend (editor).
	 *
	 * @return boolean
	 */
	public static function is_editing_block_on_backend() {
		return defined('REST_REQUEST') && true === REST_REQUEST && 'edit' === filter_input( INPUT_GET, 'context', FILTER_SANITIZE_STRING );
	}
```

**Comment by talldan** [comment]
Dropping a link to the slack conversation on this matter:
https://wordpress.slack.com/archives/C02QB2JS7/p1594147718148000

**Comment by mcsf** [comment]
> Disable form fields in the backend, preventing accidental submission.

A more general (and safer) solution is to use the `Disabled` component, often seen in Core wrapped around server-rendered widgets. For example, Archives:

https://github.com/WordPress/gutenberg/blob/33480fa7068784cd46ce92cc4d9892b535b0b271/packages/block-library/src/archives/edit.js#L36-L41

**Comment by paulschreiber** [comment]
@mcsf That worked for one of my use cases. Thanks.

**Comment by gregrobsonoxb** [comment]
If you're just looking for a simple way to change the rendered output in the backend use `is_admin()` ... worked for my needs.

**Comment by r-chrzan** [comment]
@gregrobsonoxb yes, but the problem is when we save the post because the callback function is called when saving
